src/common/com/intellij/plugins/haxe/util/HaxeNameSuggesterUtil.java

```diff
@@ -41,34 +44,225 @@ private static String deleteNonLetterFromString(@NotNull final String string) {
+    //}
+    //while (name.startsWith("_") && name.length() > 1) {
+    //  name = name.substring(1);
+    //}
+    //while (name.endsWith("_") && name.length() > 1) {
+    //  name = name.substring(0, name.length() - 1);
+    //}
+    //return name;
+
+    StringBuilder prepped = new StringBuilder();
+    int startPos = 0;
+    int endPos = name.length() - 1;
+
+    // Trim (skip past) underscores and common leading words.
+    // (Leave an underscore or name if that is the entirety of the remaining text.)
+    while (startPos < endPos && '_' == name.charAt(startPos)) ++startPos;
+    while (endPos > startPos && '_' == name.charAt(endPos)) --endPos;
+    for (String prefix : new String[]{"get", "is"}) {
+      if (name.startsWith(prefix) && (endPos - startPos) > prefix.length()) {
+        startPos += prefix.length();
+      }
+    }
+
+    // Copy the string, removing consecutive underscores.
+    char c = '_';  // Assume last char is '_' to skip any underscores after 'get' or 'is', as in get_myVar();
+    for (int i = startPos; i <= endPos; i++) {
+      if (c == '_' && name.charAt(i) == '_') continue;
+      c = name.charAt(i);
+      prepped.append(c);
+    }
+
+    return prepped.toString();
+  }
+
+
+  @NotNull
+  public static Collection<String> generateNames(@NotNull String name, boolean useUpperCase, boolean isArray) {
+    name = prepareNameTextForSuggestions(name);
+
+    Collection<String> candidates = new LinkedHashSet<String>();
+    if (null != name && !name.isEmpty()) {
+      candidates.addAll(NameUtil.getSuggestionsByName(name, "", "", useUpperCase, false, isArray));
+    }
+    return candidates;
+  }
+
+  @NotNull
+  public static String getDefaultExpressionName(HaxeExpression expression, boolean useUpperCase) {
+    String lower = getDefaultExpressionName(expression);
+    return useUpperCase ? lower.toUpperCase() : lower;
+  }
+
+  @NotNull
+  public static String getDefaultExpressionName(HaxeExpression expression) {
+    ResultHolder typeResult = HaxeTypeResolver.getPsiElementType(expression, new HaxeGenericResolver());
+    SpecificTypeReference type = typeResult.getType();
+
+    if (type.isDynamic()) { return "dyn"; }
+    if (type.isVoid()) { return "v"; }
+    if (type.isInt()) { return "i"; }
+    if (type.isBool()) { return "b"; }
+    if (type.isFloat()) { return "f"; }
+    if (type.isString()) { return "str"; }
+    if (type.isArray()) { return "arr"; }
+    if (type.isMap()) { return "map"; }
+
+    if (type instanceof SpecificHaxeClassReference) {
+      SpecificHaxeClassReference ref = (SpecificHaxeClassReference)type;
+      HaxeClass clazz = ref.getHaxeClass();
+      String name = null == clazz ? null : clazz.getName();
+      if (null != name) {
+        return HaxeStringUtil.toLowerFirst(name);
+      }
+    }
+
+    // If result typing doesn't work (e.g. it's Unknown or Invalid), then try against
+    // the kind of expression.
+
+    if (expression instanceof HaxeAssignExpression
```

Comment:
I think all default names could be moved to the plugin setting.

Reply:
Such an innocent little suggestion.  :smiley:  That will take another week.

That said, I'm not so sure that it's useful for a user, since they are likely to pick something more descriptive from the dropdown that usually occurs.  I'm not opposed to somebody filing that as an enhancement, though...

Reply:
Sure, it's a lot of work. But I guess it could be good point for the future.